### DISCUSSION TITLE
How do you expose Docker apps to the public internet?

### DISCUSSION THREAD

**Discussion by Joooost** [IceWhaleTech/CasaOS#178] [Polls]

**Comment by JohnGuan** [comment]
I'm using ZeroTier 😂

**Comment by Protektor-Desura** [comment]
If you are talking about setting this up for the typical home user and then having apps/dockers from who knows where, then doing everything from behind a VPN is going to be the simplest and most secure method. Having to configure every app/service for a reverse proxy is going to be beyond most home users.

It also means you will have to provide the reverse proxy config for every app you have in your store and it means a lot of users who install a 3rd party non-store app are going to be completely confused why somethings work and others don't work that they added when a reverse proxy is in use.

**Reply by Joooost** [reply]
Totally agree, but what is the typical home user? I believe these kind of projects, including CasaOS, Home Assistant, Vaultwarden and the maybe the majority of opensource projects, does require some IT experience and does have a steep learning curve. Most of the time you will need to work with Linux, CLI, JSON/YAML, Docker, etc. 

The lack of a customer friendly eco system and onboarding processes is covered by an enthusiastic community which will help. But in the end maybe push the typical home user away. In that case they will require to use paid software/services and that is also fine.

On topic: Wouldn't be great to embed a reverse proxy or VPN service in CasaOS  and make's the onboarding/setup more user friendly? Idea for this is to add some guided setup steps to setup VPN or Reverse Proxy.

**Reply by Protektor-Desura** [reply]
Part of the reason for me to use Dockers is I don't have to dig in to figuring out the complete install and setup of servers. Someone else has already done the majority of the heavy lifting. I would think most CasaOS users are going to assume the same. They just want stuff that just works.

**Reply by ivanswip** [reply]
VPN+nginx proxy manager
